app/src/main/java/com/example/advanceduiapp/data/MultipleViewRecyclerAdapter.kt

```diff
@@ -0,0 +1,71 @@
+package com.example.advanceduiapp.data
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import androidx.recyclerview.widget.RecyclerView.*
+import com.example.advanceduiapp.databinding.ButtonItemBinding
+import com.example.advanceduiapp.databinding.FragmentPagerBinding
+import com.example.advanceduiapp.databinding.TvItemBinding
+
+class MultipleViewRecyclerAdapter(
+    private val pageData: List<Pair<String, Int>>,
+    private val textData: List<String>,
+    private val buttonActionsData: List<() -> Unit>
```

Comment:
It's better to use here only one list of data. For example, you will have an empty abstract class(or sealed class) `Item` which will be inherited by your classes. In that case you will be able to have only one list of items